build-and-push.sh

```diff
@@ -1,48 +1,70 @@
 #!/bin/bash
 set -eu -o pipefail
 
+# set "$0" so that tmux shows something more useful than "bash" in the statusbar
 desired0='b-a-p'
 if ! grep -q "$desired0" "/proc/$$/cmdline"; then
 	exec -a "$desired0" "$SHELL" -- "$BASH_SOURCE" "$@"
 fi
 
+if [ "$#" -eq 0 ]; then
+	self="$(basename "$0")"
+	cat >&2 <<-EOF
+		error: missing arguments
+		usage: $self <build args>
+		   ie: $self debian ubuntu
+	EOF
+	exit 1
+fi
+
+# allow for specifying an alternate path to "bashbrew"
 : "${BASHBREW:=bashbrew}"
 
+# normalize "$@" to be the "--uniq" versions (and deduplicate)
+# also grab the list of associated repos in explicit build order (so we can build/push grouped by repo)
 IFS=$'\n'
-set -- $("$BASHBREW" list --uniq --repos --build-order "$@")
+set -- $("$BASHBREW" list --uniq --repos "$@" | sort -u | xargs "$BASHBREW" list --uniq --repos --build-order)
+repos=( $(echo "$*" | cut -d: -f1 | xargs "$BASHBREW" list --repos --build-order) )
 unset IFS
 
-# set -- buildpack-deps; comm -13 <(bashbrew list "$@" | sort) <(bashbrew children "$@" | sort) | xargs bashbrew list --uniq | sort -u | xargs bashbrew list --uniq --build-order
+declare -A repoTags=()
+for repoTag; do
+	repo="${repoTag%%:*}"
+	repoTags[$repo]+=" $repoTag"
+done
+
 children="$("$(dirname "$BASH_SOURCE")/children.sh" "$@")"
-: "${children:=<none>}"
 
 echo
 echo
 echo "Children: (of $*)"
-echo "$children"
+echo "${children:-<none>}"
 echo
 echo
 
-for repo; do
+for repo in "${repos[@]}"; do
+	tags=( ${repoTags[$repo]} )
 	(
 		set -x
-		time "$BASHBREW" build "$repo"
-		time "$BASHBREW" tag "$repo"
-		time "$BASHBREW" push "$repo"
+		time echo "$BASHBREW" build "${tags[@]}"
+		time echo "$BASHBREW" tag "${tags[@]}"
+		time echo "$BASHBREW" push "${tags[@]}"
```

Comment:
Facepalm.

So many times I did `--amend` and was careful to exclude that...  All it takes is once, I suppose.  Will update shortly.